### DISCUSSION TITLE
Any Pythonistas want to work on a Hyprland status bar?

### DISCUSSION THREAD

**Discussion by smsteward** [hyprwm/Hyprland#4462] [General]
Hey everyone. Was going to post this under the "contrib" repo, but it didn't have a Discussions page, so figured here was the next best place.

As a side hobby, I've started working on a Python/Gtk4 status bar in the spirit of Waybar with Hyprland in mind. I love Waybar and daily drive it, and I'm blown away at how far it's come, particularly with the Hyprland specific modules, but recently I've noticed a large number of bugs piling up that go completely unnoticed, config-breaking changes with no associated warning/changelog, and some general code inconsistencies (ok, that might be more of a me-thing, I'm a stickler for consistency, sometimes to a fault). Plus, with Gtk3 being slowly phased out in favor of Gtk4, I figured now might be a good a time as any to start working on it.

I was originally going to do it just for myself, but figured I'd poll the Hyprland community and see if there were any other Python people here that might be interested in making this more than just a personal project. If so, I'd love to know your thoughts on working on a new status bar just for Hyprland.

I've got a _few_ backend functions written for config parsing and the like, and have a _very_ basic template for drawing a window and connecting it with `libadwaita`, but that's about it. I'm a full-time business analyst/SQL dev, so my free time is mostly limited to weekends (plus it's cold where I am so my motivation has taken a hit).

If anyone is interested in potentially working on something, feel free to let me know. I haven't done much with Python since grad school, so I've been meaning to scratch this itch for a while, and figured I'd help out the community while I'm at it. Plus, I do love some jolly cooperation (I play too much Elden Ring help me).

**Comment by Xag8031** [comment]
I think the idea is good and I'm down to see new projects and help when I can. Although we already have multiple options, so what do you have planned that makes it better than waybar or any other status bar?

**Reply by smsteward** [reply]
That's a great question, and my honest answer is, I'm not really sure. Step 1 I guess would be to get a stable code base and port some of the core Waybar modules over, and then work on adding features. Like I said, Waybar serves my (admittedly rather simple) use-case right now, but a lot of bugs are piling up that receive no attention, and more than a few PR's have been merged recently that immediately break a rather basic feature. I'd ideally like to have a stable code base based on the latest toolkit, and a few core members that independently verify and sign-off on PRs before merging to make sure features don't break things, like we do in my workplace.

So, I guess the tl;dr is, I don't have any "killer" features planned at the moment that would put the existing options on a lower tier. But the lofty end goal is something dedicated to hyprland that is based on latest toolkits and tries to not break things.
